shutdown.go

```diff
@@ -60,6 +60,9 @@ func (app *App) broadcastSignal(signal os.Signal) error {
 	app.donesMu.RLock()
 	defer app.donesMu.RUnlock()
 
+	// Record a shutdown signal
+	app.shutdownSig = signal
```

Comment:
This isn't thread-safe. We're modifying state with a read-lock. We need a write lock if we're modifying state, or we can use an atomic.